bootstrap/bootstrap.go

```diff
@@ -1216,6 +1216,10 @@ func (b *Bootstrap) updateGitMirror() (string, error) {
 		b.shell.Commentf("Cloning a mirror of the repository to %q", mirrorDir)
 		flags := "--mirror " + b.GitCloneMirrorFlags
 		if err := gitClone(b.shell, flags, b.Repository, mirrorDir); err != nil {
+			b.shell.Commentf("Removing mirror dir %q due to failed clone", mirrorDir)
+			if err := os.RemoveAll(mirrorDir); err != nil {
```

Comment:
i suspect this will cause issues with the lockfile we're creating above on [line 1208](https://github.com/buildkite/agent/pull/1671/files#diff-70e82e7432e3056b7466bb3e4fe16874c12cfbe7bbb71a15ad9d1997217e94c9R1208) - if we delete the lockfile before we're finished deleting everything else, something sharing the mirror dir might run into a very weird situation

Reply:
The lockfile is a peer of the mirror dir I think, not a child:

https://github.com/buildkite/agent/blob/1376fb737e8e33d83bd47d6404d8cc2cca458c45/bootstrap/bootstrap.go#L1208

So should be fine to return an error and run the deferred unlock, right?

Reply:
oh you're completely right, i hallucinated a `/` in there. as you were!